### ISSUE TITLE
Create important branches and update branch protection rules

### ISSUE THREAD

**Issue by zhannaklimanova** [DDMAL/VIM#6]
- [ ] develop (default branch)
- [ ] staging (maybe)
- [ ] production (maybe)
- [ ] main (maybe)

**Comment by dchiller** [comment]
Any additional thoughts on this? I protected main from direct pushes last week, but looking like that didn't work exactly, so gonna try to figure out why.

**Comment by zhannaklimanova** [comment]
> Any additional thoughts on this? I protected main from direct pushes last week, but looking like that didn't work exactly, so gonna try to figure out why.

Yup, so we're going to need to look into this more because I initially thought there would be an automatic PR when pushing to develop and when I recently pushed my README changes, they were merged right away. I thought that there were stricter branch protection rules and that a PR would be created for me automatically to merge develop into develop but that didn't happen. It's just the markdown that I modified so nothing catastrophic got merged but this is still something we will need to modify. I will do some testing on this over the week and see what I can come up with as a solution.

**Comment by dchiller** [comment]
> Yup, so we're going to need to look into this more because I initially thought there would be an automatic PR when pushing to develop and when I recently pushed my README changes, they were merged right away

I just disabled force pushes for the `develop` branch. For a PR though, you shouldn't push to the develop branch, because a PR is between different branches.

**Comment by zhannaklimanova** [comment]
> What branch would the PR be from if you are pushing to the same branch as the branch the PR is to?

I believe I’ve seen somewhere a PR that had develop->develop. The idea is that after we have pulled and rebased our changes locally, when pushing to the remote version of develop, there is just a PR issued as a final check before allowing those changes to be integrated into the global repository code. I don’t know if this is at all possible however.

**Comment by dchiller** [comment]
> I believe I’ve seen somewhere a PR that had develop->develop

I suspect those are from different forks. if you have your own fork and you push to the develop branch on that fork, you can then PR into the develop branch on DDMAL.

**Comment by zhannaklimanova** [comment]
Oh, yes that might be the case. Thanks for clarifying. 

> > I believe I’ve seen somewhere a PR that had develop->develop
> 
> I suspect those are from different forks. if you have your own fork and you push to the develop branch on that fork, you can then PR into the develop branch on DDMAL.

**Comment by dchiller** [comment]
This is complete. `develop`, `staging`, and `production` exist, and all have appropriate branch protection rules (PR with approvals required)
